lib/ballot.go

```diff
@@ -311,12 +326,19 @@ func (b *BallotBoxes) IsVoted(ballot Ballot) bool {
 	return vr.IsVoted(ballot)
 }
 
-func (b *BallotBoxes) AddVotingResult(vr *VotingResult, bb *BallotBox) (err error) {
+func (b *BallotBoxes) AddVotingResult(vr *VotingResult, ballot Ballot) (err error) {
 	b.Lock()
 	defer b.Unlock()
 
 	b.Results[vr.MessageHash] = vr
-	bb.AddVotingResult(vr) // TODO detect error
+
+	if ballot.CanFitInVotingBox() {
+		err = b.VotingBox.AddVotingResult(vr)
+	} else if ballot.CanFitInWaitingBox() {
+		err = b.WaitingBox.AddVotingResult(vr)
+	} else {
+		// do nothing
```

Comment:
It can not be possible, not `ballot.CanFitInVotingBox()` and not `ballot.CanFitInWaitingBox()`, so how about using `log.Warn` to detect this weird case in logs.